### DISCUSSION TITLE
Use of `delay` outside of `progressive_val_score`

### DISCUSSION THREAD

**Discussion by cdeterman** [online-ml/river#1621] [General]
The use of the `delay` argument within the `progressive_val_score` function is very useful, however, what if I need to do any additional processing, storing multiple metrics, or returning the model itself (or is the `model` updating in-place when it is passed)?  Is there a way to implement `delay` directly in the normal loop with `predict_one`/`learn_one`?

**Comment by MaxHalford** [comment]
Hey there. I'm not sure this fully answers your question, but under the hood `progressive_val_score` is using [`stream.simulate_qa`](https://riverml.xyz/latest/api/stream/simulate-qa/). The latter should give you more control over the fit/predict loop.
